modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -55,10 +55,10 @@ object BlockService {
           addressStorage.getBalance(address).map(_.some)
 
         def getLastTxRef(address: Address): F[Option[transaction.TransactionReference]] =
-          transactionStorage.getLastAcceptedReference(address).map(_.some)
+          transactionStorage.getLastProcessedTransaction(address).map(_.ref.some)
```

Comment:
The name of this function seems wrong. The function returns last accepted transaction (either Majority or Accepted).

Reply:
Yes.
Transaction has the following lifecycle:

Waiting -> Processing -> Accepted -> Majority

so Processed means either Accepted or Majority or anything which comes after Processing